src/charm.py

```diff
@@ -344,6 +345,41 @@ def _on_install(self, _event: InstallEvent) -> None:
         """
         state = self._setup_state()
 
+        if state.charm_config.openstack_clouds_yaml:
```

Comment:
The changes on charm.py blocks on all hooks on openstack configuration - other hooks have dependencies, e.g. on_config_changed depends on on_install hook to run properly.